### PR TITLE
Remove old demo code

### LINKED ISSUE #31: Remove old demo
https://github.com/dwave-examples/structural-imbalance/pull/25 added a new implementation of the demo but did not go so far as to clean up the old one.

### PR DESCRIPTION
**Pull request by mcfarljm**
Remove `demo.py` in favor of new driver code `structural_imbalance.py`.  Update `test_integration.py` accordingly (and add some checks that conditionally set `matplotlib.use("agg")` to avoid issues testing on headless systems, given that the new code imports `pyplot`).  Remove `test_interfaces.py`, which only tested interface code that was used by `demo.py` but not `structural_imbalance.py`.  Remove everything under `dwave_structural_imbalance_demo` that was specific to `demo.py` (everything except for `mmp_network`).  Update README accordingly to reflect new usage and output.

The functionality of the new driver script is different than the previous one.  In particular, it does not generate the CSV output or the date-specific plot files.  Please confirm this is what we want.

Closes #31.

### CODE REVIEW

File: structural_imbalance.py [hunk 1]
```diff
@@ -12,17 +12,25 @@
...
 import click
 import matplotlib
-import matplotlib.pyplot as plt
+# Trap errors with importing pyplot (for testing frameworks) and
+# specify "agg" backend
+try:
+    import matplotlib.pyplot as plt
+except ImportError:
+    matplotlib.use("agg")
+    import matplotlib.pyplot as plt
```
Comment (randomir) [inline structural_imbalance.py:25]: Nice workaround.

**Review by randomir — APPROVED** [review]
Looks great. I would just remove the package files (manifest, setup.py). Also requirements should be updated, but that can be a separate PR.

File: drawing.py [hunk 2]
```diff
@@ -12,6 +12,8 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import sys
```
Comment (arcondello) [inline drawing.py:15]: Why did these come in? I don't see where they are used

File: mmp_network/__init__.py [hunk 3]
```diff
@@ -0,0 +1 @@
+from .loader import *
```
Comment (arcondello) [inline mmp_network/__init__.py:1]: IMO better to use absolute import

**Review by arcondello — APPROVED** [review]
Net -711 lines! Love it

File: drawing.py (hunk 2 above)
Comment (mcfarljm) [reply drawing.py:15]: Good call, it got introduced in 5973ca76 with my first approach of handling the pyplot issue by checking for `sys.platform == linux`, but then I forgot to remove it when I switched to a different approach.  Will fix.

**Comment by mcfarljm** [comment]
Thanks for the reviews, should be ready to merge

**Pull request merged**